enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/messaging/SenderService.java

```diff
@@ -78,12 +75,17 @@ public void send( AdvertisedSocketAddress to, Message message )
                 return;
             }
 
-            channel( to ).send( message );
+            future = channel( to ).send( message );
         }
         finally
         {
             serviceLock.readLock().unlock();
         }
+
+        if ( block )
+        {
+            future.awaitUninterruptibly();
+        }
```

Comment:
Good observation and I actually was going for that initially but decided to just contain it with a simple block flag. I think there is a lot of potential for future maturity to be gained in the messaging stack. I am trying to be minimally invasive here.